src/Sentry/SentryOptionsExtensions.cs

```diff
@@ -236,5 +237,30 @@ internal static void SetupLogging(this SentryOptions options)
                 options.DiagnosticLogger = null;
             }
         }
+
+        /// <summary>
+        /// By default, the StartupTime is set once SentryOptions is created.
+        /// That way is computationally cheap but not so precise.
+        /// You call this extension to give a better precision to the StartupTime at a cost of a small overhead to current
+        /// Thread.
+        /// </summary>
+        /// <param name="options"></param>
+        internal static void SetupStartupTime(this SentryOptions options)
+        {
+            _ = Task.Run(() =>
+            {
+                try
+                {
+                    using (var proc = Process.GetCurrentProcess())
+                    {
+                        options.StartupTime = proc.StartTime.ToUniversalTime();
+                    }
+                }
+                finally
+                {
+                    //Ignore any exception and stay with the fallback DateTime.UtcNow value.
+                }
+            }).ConfigureAwait(false);
+        }
```

Comment:
Why do we need it on `SentryOptions`?
Can't we just have it lazily initialized on `Contexts.App` instead?

Reply:
Also not a fan of extending SentryOptions, doesn't belong here.

Could be its own internal class, something like `ProcessInfo`.

It can have a property `DateTimeOffset StartupTime` that starts off with the `DateTimeOffset.UtcNow` (set inlined to the propert decalaration) but does the Task.Run in the  constructor to set the "better" value afterwards.

Please add a catch block and suppress all exceptions (so it doesn't go into UnobservedTaskException), ideally logging with `options.Logger?.LogError`.

You can add a `static ProcessInfo` field to `SentrySdk` that gets set when `Init` is called (so the UtcNow is read as early as possible: i.e: Now when the first event goes out) and increases the chance that an error early in the process uses the "worst" value.

Lastly, this gives a chance of introducing an opt-out flag on `SentryOptions` later if this becomes a problem (Task.Run on Unity or something), so we can check for this flag before instantiating this class during SentrySdk.Init.

Reply:
I wonder why we need the task at all, is it not possible to just use `Lazy<>`?

```csharp
var startupTime = new Lazy<DateTimeOffset>(() =>
{
      using var proc = Process.GetCurrentProcess();
      return proc.StartTime.ToUniversalTime();
});
```

Reply:
How would Lazy work in this scenario?
StartupTime is initially set with UtcNow. After that, it'll be set only once with a more precise Timer.

The task is there to save some ms from the calling thread since GetCurrentProcess takes ~80ms to be executed.
I'd agree that Lazy would make sense if StartupTime was only set with proc.StartTime.ToUniversalTime, but we are initializing it with DateTimeOffset.UtcNow and also the Process StartTime.